entity-framework/core/providers/cosmos/querying.md

```diff
@@ -0,0 +1,312 @@
+---
+title: Querying - Azure Cosmos DB Provider - EF Core
+description: Querying with the Azure Cosmos DB EF Core Provider
+author: roji
+ms.date: 09/19/2024
+uid: core/providers/cosmos/querying
+---
+# Querying with the EF Core Azure Cosmos DB Provider
+
+## Querying basics
+
+[EF Core LINQ queries](xref:core/querying/index) can be executed against Azure Cosmos DB in the same way as for other database providers. For example:
+
+```csharp
+public class Session
+{
+    public Guid Id { get; set; }
+    public string Category { get; set; }
+
+    public string TenantId { get; set; } = null!;
+    public Guid UserId { get; set; }
+    public int SessionId { get; set; }
+}
+
+var stringResults = await context.Sessions
+    .Where(
+        e => e.Category.Length > 4
+            && e.Category.Trim().ToLower() != "disabled"
+            && e.Category.TrimStart().Substring(2, 2).Equals("xy", StringComparison.OrdinalIgnoreCase))
+    .ToListAsync();
+```
+
+> [!NOTE]
+> The Azure Cosmos DB provider does not translate the same set of LINQ queries as other providers.
+> For example, the EF `Include()` operator isn't supported on Cosmos, since cross-document queries aren't supported in the database.
+
+## Partition keys
+
+The advantage of partitioning is to have your queries execute only against the partition where the relevant data is found, saving costs and ensuring faster result speed. Queries which don't specify partition keys are executed on all the partitions, which can be quite costly.
+
+Starting with EF 9.0, EF automatically detects and extracts partition key comparisons in your LINQ query's `Where` operators. Let's assume we execute the following query against our `Session` entity type, which is configured with a hierarchical partition key:
+
+```csharp
+protected override void OnModelCreating(ModelBuilder modelBuilder)
+{
+    modelBuilder.Entity<Session>()
+        .HasPartitionKey(b => new { b.TenantId, b.UserId, b.SessionId })
+}
+
+var tenantId = "Microsoft";
+var userId = new Guid("99A410D7-E467-4CC5-92DE-148F3FC53F4C");
+var username = "scott";
+
+var sessions = await context.Sessions
+    .Where(
+        e => e.TenantId == tenantId
+             && e.UserId == userId
+             && e.SessionId > 0
+             && e.Username == username)
+    .ToListAsync();
+```
+
+Examining the logs generated by EF, we see this query executed as follows:
+
+```sql
+Executed ReadNext (166.6985 ms, 2.8 RU) ActivityId='312da0d2-095c-4e73-afab-27072b5ad33c', Container='test', Partition='["Microsoft","99a410d7-e467-4cc5-92de-148f3fc53f4c"]', Parameters=[]
+SELECT VALUE c
+FROM root c
+WHERE ((c["SessionId"] > 0) AND CONTAINS(c["Username"], "a"))
+```
+
+In these logs, we notice the following:
+
+* The first two comparisons - on `TenantId` and `UserId` - have been lifted out, and appear in the ReadNext's "Partition" rather than in the `WHERE` clause; this means that query will only execute on the subpartitions for those values.
```

Comment:
```suggestion
* The first two comparisons - on `TenantId` and `UserId` - have been lifted out and appear in the `ReadNext` "Partition" rather than in the `WHERE` clause; this means that query will only execute on the subpartitions for those values.
```